pool.go

```diff
@@ -0,0 +1,166 @@
+package viamrtsp
+
+import (
+	"sync"
+	"time"
+
+	"go.viam.com/rdk/logging"
+)
+
+type poolItem struct {
+	frameWrapper *avFrameWrapper
+	lastAccess   time.Time
+}
+
+type framePool struct {
+	items          []poolItem
+	maxAge         time.Duration
+	preseededCount int
+
+	putCount   int
+	getCount   int
+	cleanCount int
+	newCount   int
+
+	mu           sync.RWMutex
+	stopCleaning chan struct{}
+	logger       logging.Logger
+}
+
+func newFramePool(maxAge, cleanupInterval time.Duration, preseededCount int, logger logging.Logger) *framePool {
+	pool := &framePool{
+		items:          make([]poolItem, 0, preseededCount),
+		maxAge:         maxAge,
+		preseededCount: preseededCount,
+		stopCleaning:   make(chan struct{}),
+		logger:         logger,
+	}
+
+	// Pre-seed the pool
+	for i := 0; i < preseededCount; i++ {
+		pool.items = append(pool.items, poolItem{
+			frameWrapper: pool.new(),
+			lastAccess:   time.Now(),
+		})
+	}
+
+	go pool.cleanupRoutine(cleanupInterval)
+
+	return pool
+}
+
+func (p *framePool) new() *avFrameWrapper {
+	p.logger.Debug("newFunc for pool was called!")
+	avFrame, err := newAVFrameWrapper()
+	if err != nil {
+		p.logger.Errorf("Failed to allocate AVFrame in frame pool: %v", err)
+		return nil
+	}
+	p.newCount++
+	return avFrame
+}
+
+func (p *framePool) get() *avFrameWrapper {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+
+	if len(p.items) == 0 {
+		return nil
+	}
+
+	item := p.items[0]
+	p.items = p.items[1:]
+
+	p.logger.Debugf("Item was gotten from the pool. Len now: %d", len(p.items))
+	p.getCount++
+
+	item.frameWrapper.isInPool.Store(false)
+	return item.frameWrapper
+}
+
+func (p *framePool) put(frame *avFrameWrapper) {
```

Comment:
The surface area of this API is super small, this is fine -- not mandating a change here. 

But if users should* be calling `safelyPut` -- that should be named `put` to be more discoverable/the thing a user is most likely to type. And this can maybe be `doPut`? But honestly, I'd instead De Morgan the conditional in `safelyPut` and have the error case early-return. And I'd just move this logic inline with the "safely put" logic. Letting us delete this function.

Reply:
Consolidated into `put`. Deleted `safelyPut`